cli/src/generate/prepare_grammar/expand_tokens.rs

```diff
@@ -167,6 +167,11 @@ impl NfaBuilder {
                 }
                 Ok(true)
             }
+            Rule::Immediate(_next_rule) => {
+                self.is_sep = false;
+                self.push_advance(CharacterSet::empty(), next_state_id);
+                Ok(true)
```

Comment:
Here we need to somehow instruct the parser to not `SKIP` and immediately `ADVANCE` to the next rule.